### PR TITLE
Fix various issues around missing engine functions and coercions.

### PR DESCRIPTION
**Pull request by mcovarr**

### CODE REVIEW

**Comment by mcovarr** [comment]
Addresses float-related issues discovered by @bradtaylor.

File: src/main/scala/cromwell/engine/backend/local/LocalEngineFunctions.scala [hunk 1]
```diff
@@ -64,7 +64,19 @@ class LocalEngineFunctions(cwd: Path, stdout: Path, stderr: Path) extends WdlSta
...
+   * Try to read a float from the file referenced by the specified `WdlValue`.
+   */
+  override protected def read_float(params: Seq[Try[WdlValue]]): Try[WdlFloat] =
+    read_string(params) map { s => WdlFloat(s.value.trim.toDouble) }
+
+  /**
+   * Try to read a boolean from the file referenced by the specified `WdlValue`.
+   */
+  override protected def read_boolean(params: Seq[Try[WdlValue]]): Try[WdlBoolean] =
+    read_string(params) map { s => WdlBoolean(java.lang.Boolean.parseBoolean(s.value.trim.toLowerCase)) }
```
Comment (scottfrazer) [inline src/main/scala/cromwell/engine/backend/local/LocalEngineFunctions.scala]: Is this implemented on the JES backend?

Comment (mcovarr) [inline src/main/scala/cromwell/engine/backend/local/LocalEngineFunctions.scala]: No.  Actually I should refactor read_int/read_float/read_boolean up to the trait.

**Comment by scottfrazer** [comment]
:+1:

**Comment by scottfrazer** [comment]
@geoffjentry second reviewer?

**Comment by geoffjentry** [comment]
:100: / :100:

**Pull request merged**